src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useContext, useState } from 'react';
+import { TodoContext } from '../../context/TodoContext';
+
+import { NewTodo } from '../NewTodo';
+import { TodoList } from '../TodoList';
+import { TodoFooter } from '../TodoFooter';
+
+import { Filter } from '../../types/Filter';
+
+export const TodoApp = () => {
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const { todos, setTodo } = useContext(TodoContext);
+
+  const todosAllCompleted = todos.filter(todo => !todo.completed).length === 0;
```

Comment:
It is recommended to memoize the 'todosAllCompleted' calculation to avoid unnecessary recalculations during re-renders.